core/app/models/spree/tax_rate.rb

```diff
@@ -142,8 +142,8 @@ def potentially_applicable?(order_tax_zone)
       self.zone == order_tax_zone ||
       # If the rate's zone *contains* the order's tax zone, then it's applicable.
       self.zone.contains?(order_tax_zone) ||
-      # 1) The rate's zone is the default zone, then it's always applicable.
-      (self.included_in_price? && self.zone.default_tax)
+      # 1) The rate's zone is in the default zone, then it's always applicable.
+      (self.included_in_price? && self.zone.contains?(Spree::Zone.default_tax))
```

Comment:
Yes, correct, the comment is not correct in this case.

There is this idea of a smallest possible zone (`Spree::Zone.match(address)`) produces just that, the smallest fitting zone for an address. Apparently, a default zone should also be like that (contain no other zones). 

I'll fix the comment.